packages/core/src/api/middleware/auth-guard.ts

```diff
@@ -59,7 +63,10 @@ export class AuthGuard implements CanActivate {
         if (targetIsFieldResolver) {
             requestContext = internal_getRequestContext(req);
         } else {
-            const session = await this.getSession(req, res, hasOwnerPermission);
+            const apiKeyHashingStrategy = this.apiKeyService.getApiKeyAuthOptionsByApiType(
+                getApiType(info),
+            ).hashingStrategy;
+            const session = await this.getSession(req, res, hasOwnerPermission, apiKeyHashingStrategy);
```

Comment:
We need to be able to handle the scenario where the API Key session got deleted for some reason (eg manual "clean up" of session table)